src/main/java/io/gravitee/common/util/EnvironmentUtils.java

```diff
@@ -93,6 +98,70 @@ public static String encodedKey(String key) {
         return result;
     }
 
+    public static boolean hasMatchingTags(Optional<List<String>> configuredTags, Set<String> tags) {
+        if (configuredTags.isPresent()) {
+            List<String> tagList = configuredTags.get();
+            if (tags != null) {
+                final List<String> inclusionTags = tagList
+                    .stream()
+                    .map(String::trim)
+                    .filter(tag -> !tag.startsWith("!"))
+                    .collect(Collectors.toList());
+
+                final List<String> exclusionTags = tagList
+                    .stream()
+                    .map(String::trim)
+                    .filter(tag -> tag.startsWith("!"))
+                    .map(tag -> tag.substring(1))
+                    .collect(Collectors.toList());
+
+                if (inclusionTags.stream().anyMatch(exclusionTags::contains)) {
+                    throw new IllegalArgumentException("You must not configure a tag to be included and excluded");
+                }
+
+                return (
+                    (
+                        inclusionTags.isEmpty() ||
+                            inclusionTags
```

Comment:
suggestion : could we avoid duplication by using a function like `tagsContains(Set<String> tags, List<String> searchedTags)`?

and just use it like : `return tagsContains(tags, inclusionTags) && !tagContains(tags, exclusionTags)`